### PR TITLE
[mlir][mesh] Better Op result names

### PR DESCRIPTION
**Pull request by sogartar**
Implement OpAsmOpInterface for most ops to increase IR readability. For example `mesh.process_linear_index` would produce a value with name `proc_linear_idx`.

### CODE REVIEW

File: mlir/lib/Dialect/Mesh/IR/MeshOps.cpp [hunk 1]
```diff
@@ -180,6 +182,20 @@ Type mesh::shardType(Type type, MeshOp mesh, MeshShardingAttr sharding) {
...
+//     return;
+//   }
+//   SmallString<64> str;
+//   for (auto [i, result]: llvm::enumerate(op->getResults())) {
+//     (Twine(namePrefix) + "_" + Twine(i) + "_").toStringRef(str);
+//     setNameFn(result, str);
+//     str.clear();
+//   }
+// }
+
```
Comment (joker-eph) [inline mlir/lib/Dialect/Mesh/IR/MeshOps.cpp:198]: Remove?

File: mlir/lib/Dialect/Mesh/IR/MeshOps.cpp [hunk 2]
```diff
@@ -27,7 +27,9 @@
 #include "llvm/ADT/DenseSet.h"
 #include "llvm/ADT/STLExtras.h"
 #include "llvm/ADT/SmallSet.h"
+#include "llvm/ADT/SmallString.h"
 #include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/Twine.h"
```
Comment (joker-eph) [inline mlir/lib/Dialect/Mesh/IR/MeshOps.cpp:32]: Nit: these includes don't seem needed from the diff

**Review by joker-eph — APPROVED** [review]

File: mlir/lib/Dialect/Mesh/IR/MeshOps.cpp (hunk 2 above)
Comment (sogartar) [reply mlir/lib/Dialect/Mesh/IR/MeshOps.cpp]: I cleaned up the unused headers.

**Review by joker-eph — APPROVED** [review]

**Comment by sogartar** [comment]
@joker-eph, thank you for your timely review.

**Pull request merged**